src/types.ts

```diff
@@ -68,3 +68,9 @@ export type InputAmount = {
     decimals: number;
     rawAmount: bigint;
 };
+
+export type InitInputAmount = InputAmount | WeightedInitInputAmount;
+
+export type WeightedInitInputAmount = InputAmount & {
```

Comment:
I'd follow the reversed pattern we've been using:
- `InitInputAmount` -> `InputAmountInit`
- `WeightedInitInputAmount` -> `InputAmountWeighted`